elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java

```diff
@@ -400,6 +400,21 @@ public static void setupInstance() throws Exception {
             "aggregations:{'v1.max.field': 'val1'",
             "'v2.min.field': 'val2'}"))
         .returnsUnordered("v1=7; v2=5");
+  }
+
+  @Test void testNullsSort() {
```

Comment:
Can we remove this tests out the AggregationTest ? This unit test isn't belong to the type.

Reply:
I've thought about adding NULLS ORDER tests in ElasticSearchAdapterTest, which is more reasonable.
However, many tests are disabled after CALCITE-4101(see https://github.com/apache/calcite/commit/e7aca6914c65bd812a9b177c438f687b77e090af), as a result of CI failure.
I think it's better to recover those test cases. What do you suggest? @NobiGo

Reply:
@ILuffZhe Re-enabling the tests that pass sounds a reasonable thing to do.

Reply:
@zabetak OK, I am gonna to file a JIRA case to document it. Test case for this PR shall be adjusted after that.

Reply:
+1 Enabling the tests.

Reply:
Hi, @NobiGo , @zabetak , I need a little advice on this patch.
I'm gonna to add some unit tests for NULLS SORT, however, ElasticSearchAdapterTest use zips-mini.json as its data, which doesn't contain any null values for each column, so adding or deleting data would cause a lot code changes. I've thought about two ways:

- add test cases in ElasticSearchAdapterTest without changing zips-mini.json, only check the generated Elasticsearch Scripts 
- add a new class for NULLS SORT, kinda of redundant since there are several test classes in this module(if any new features cannot be testified based on the existing tests, always adding a class might be unacceptable)

What do you think?

Reply:
@ILuffZhe I don't have a strong preference.

There are already "feature" specific classes such as "ProjectionTest", "MatchTest", etc., so I wouldn't be surprised if there is a new class emphasizing on sorting and nulls.

On the other hand you could definitely add another dataset in `ElasticSearchAdapterTest` and add test cases in that class. You may draw some inspiration from `JdbcAdapterTest` where each test uses a specific `model`.

Reply:
@zabetak Thanks for the helpful hints. I'll take a look afterwards.

Reply:
I think we should add another like zips-mini-nullable.json file include NULL value in doc, then after when we need add another test for NULL value inlucde aggregate... can base on this doc. About Where the unit test should be, I don't have a strong preference.

Reply:
Thanks for the advice @NobiGo .
I've added a new dataset(zips-mini-nullable.json) for NULLS SORT, and made some adjustments on managing different datasets in ElasticSearchAdapterTest. Please take a look when you are free.

Reply:
Hi @ILuffZhe,

I agree with Stamatis that you should validate actual query similarly to how it is done in `ElasticSearchAdapterTest`.

Perhaps, you can create separate class `SortTest` (or `OrderTest`) to validate output as well as actual queries ? For examples look at `BooleanLogicTest` or `AggregationTest`.

Reply:
Hi @asereda-gs ,
I've though about creating a new class for testing NULLS SORT before, since NULLS SORT is one small sorting feature, I decided to add a dataset to verify its function at last, like @zabetak mentioned above.
What's more, for those contributors who want to add new features or fixing bugs in ES module, I think creating different indexes in one class(like ElasticSearchAdapterTest) for unit tests is also a good choice.

Reply:
Hi @asereda-gs ,
I've though about creating a new class for testing NULLS SORT before, since NULLS SORT is one small sorting feature, I decided to add a dataset to verify its function at last, like @zabetak mentioned above.
What's more, for those contributors who want to add new features or fixing bugs in ES module, I think creating different indexes in one class(like ElasticSearchAdapterTest) for unit tests is also a good choice.

Reply:
I'm not sure you need a very big dataset, just a smart one.

Personally I find it easier to add test documents (for ES) directly in java code see [AggregationTest](https://github.com/apache/calcite/blob/cdf9d700183d926780f4e5b57d6211a85a548506/elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java#L72-L74) and then run different queries against them.

Reply:
For this case, one small dataset is enough. And personally speaking, adding a new test class is much more convenient. So, your advice is reasonable.
As for why I add a new dataset in ElasticSearchAdapterTest, I think creating an index seems a little better than creating a test class when the existing data is not satisfied for fixing bugs, which require not many tests.
We might need a third opinion here. What do you think? @zabetak @NobiGo

Reply:
As above, I don't care this test cases should be in ElasticSearchAdapterTest or another new Class, It's up to you. But what I care is the PR is really address this issue and have a more comprehensive test cases.

Reply:
I think there are too many unnecessary changes in `ElasticSearchAdapterTest`. You can reuse `AggregationTest` or create your own class (eg. `SortTest`). I assume you would want to test both aggregation and sorting in the same query.

Reply:
Yeah, I want to confirm order by in aggregation terms still works.
Moreover, I'm going to create a SortTest to address this issue based on the overall discussions here. Thanks for your advice @asereda-gs .